File: AlgorithmRise/extra/LeetCode/剑指offer-LCOF/面试题27. 二叉树的镜像/draft-1.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var mirrorTree = function(root) {
    
    let result = [], nodes = [root];
    while( nodes.length ){
        
        let stack = [];
        let next_nodes = [];
        for( let i = nodes.length-1; i>=0; --i ){
            stack.unshift( nodes[i].val );
            if( nodes[i].right ){
                next_nodes.unshift(nodes[i].right);
            }
            if( nodes[i].left ){
                next_nodes.unshift(nodes[i].left);
            }
        }
        nodes = Object.assign([], next_nodes);
        result.concat( stack );
    }

    return result;
};

File: AlgorithmRise/extra/LeetCode/剑指offer-LCOF/面试题32 - I. 从上到下打印二叉树/draft-1.go
package main2

import "fmt"

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

 var nodes map[int][]int

 func traverseTree(node *TreeNode, level int) int {
 
	 if nil == node {
		 return 0
	 }
 
	 if _, ok := nodes[level]; ok{
		 nodes[level] = append(nodes[level], node.Val)
	 } else {
		 nodes = make(map[int][]int, 1005)
	 }
 
	 if nil != node.Left {
		 traverseTree(node.Left, level+1)
	 }
 
	 if nil != node.Right {
		 traverseTree(node.Right, level+1)
	 }
 
	 return 0
 }
 
 func levelOrder(root *TreeNode) []int {
 
	 // 定义一个二维数组
	 traverseTree(root, 0)
 
	 fmt.Println(nodes)
 
	 var t = []int{2, 3}
 
	 return t
 }